vxquery-core/src/main/java/org/apache/vxquery/serializer/XMLSerializer.java

```diff
@@ -341,6 +325,67 @@ private void printAttributeNode(PrintStream ps, TaggedValuePointable tvp) {
         }
     }
 
+    private void printObject(PrintStream ps, TaggedValuePointable tvp) {
+        ObjectPointable op = pp.takeOne(ObjectPointable.class);
+        SequencePointable seqp = pp.takeOne(SequencePointable.class);
+        VoidPointable vp = pp.takeOne(VoidPointable.class);
+        UTF8StringPointable utf8sp = pp.takeOne(UTF8StringPointable.class);
+        try {
+            tvp.getValue(op);
+            op.getKeys(tvp);
+            if (tvp.getTag() == ValueTag.SEQUENCE_TAG) {
```

Comment:
It seems that this case should never happen as printSequence should be used. So this if-branch should be removed.